imagery-cesium/plugins/draw-annotations.visualizer.ts

```diff
@@ -54,87 +62,75 @@ export class CesiumDrawAnnotationsVisualizer extends BaseImageryPlugin {
 
 		if (this.drawingMode === AnnotationMode.Point) {
-			return this.leftClickEvent$.pipe(take(1), map((screenPixels: IPixelPosition) => {
+			const leftClickEventSubscription = this.leftClickEvent$.pipe(take(1), tap((screenPixels: IPixelPosition) => {
 				const earthPosition = this.cesiumMap.getEarthPositionFromScreenPixels(screenPixels.position);
-				return this.generateGeometry(this.drawingMode, [earthPosition]);
-			}));
-		}
 
-		this.mapEventsSubscription.add(
-			this.leftClickEvent$.subscribe((screenPixels: IPixelPosition) => {
-				const earthPosition = this.cesiumMap.getEarthPositionFromScreenPixels(screenPixels.position);
 				if (defined(earthPosition)) {
-					if (this.activeShapePoints.length === 0) {
-						// floating point is the point that sticks to the cursor when drawing!
-						// on every mouse move I update it's value to the last mouse position
-						// here I create it for the first time
-						this.floatingPoint = this.addAnnotation(AnnotationType.Point, earthPosition);
-
-						this.activeShapePoints.push(earthPosition);
-						const dynamicPositions = new CallbackProperty(() => {
-							if (this.drawingMode === AnnotationMode.Polygon) {
-								return new PolygonHierarchy(this.activeShapePoints);
-							}
-							return this.activeShapePoints;
-						}, false);
-						this.activeShape = this.addAnnotation(this.drawingMode as unknown as AnnotationType, dynamicPositions);
-					}
+					this.activeShapePoints = [earthPosition];				
+					this.events.onDrawEnd.next(this.onDrawEnd());
+				}
+			})).subscribe();
+			this.mapEventsSubscription.add(leftClickEventSubscription);
+
+			return true;
+		}
+
+		const leftClickEventSubscription = this.leftClickEvent$.subscribe((screenPixels: IPixelPosition) => {
+			const earthPosition = this.cesiumMap.getEarthPositionFromScreenPixels(screenPixels.position);
+			if (defined(earthPosition)) {
+				if (!this.activeShapePoints) {
+					this.activeShapePoints = [];
+
 					this.activeShapePoints.push(earthPosition);
+					const dynamicPositions = new CallbackProperty(() => {
+						if (this.drawingMode === AnnotationMode.Polygon) {
+							return new PolygonHierarchy(this.activeShapePoints);
+						}
+						return this.activeShapePoints;
+					}, false);
+					this.addAnnotation(this.drawingMode as unknown as AnnotationType, dynamicPositions);
 				}
-			})
-		);
-		this.mapEventsSubscription.add(
-			this.mouseMoveEvent$.subscribe((event) => {
-				this.updateLastActivePoint(event.endPosition);
-			})
-		);
-
-		return this.leftDoubleClickEvent$.pipe(map(() => this.onDrawEnd()));
+				this.activeShapePoints.push(earthPosition);
+			}
+		})
+
+		const leftDoubleClickEventSubscription = this.leftDoubleClickEvent$.pipe(take(1), tap(() => this.events.onDrawEnd.next(this.onDrawEnd()))).subscribe();
+
+		this.mapEventsSubscription.add(leftDoubleClickEventSubscription);
+		this.mapEventsSubscription.add(leftClickEventSubscription);
```

Comment:
are you sure you don't need to keep all the subscriptions in array and unsubscribe all of them? have you checked it?

Reply:
Yes I checked it! 
"Subscriptions can also be put together, so that a call to an unsubscribe() of one Subscription may unsubscribe multiple Subscriptions. You can do this by "adding" one subscription into another" - https://rxjs-dev.firebaseapp.com/guide/subscription